src/plugins/nginx.go

```diff
@@ -427,123 +449,278 @@ func (n *Nginx) validateConfig(nginx *proto.NginxDetails, correlationId string,
 	}
 }
 
-func (n *Nginx) completeConfigApply(response *NginxConfigValidationResponse) *proto.Command_NginxConfigResponse {
+func (n *Nginx) completeConfigApply(response *NginxConfigValidationResponse) (status *proto.Command_NginxConfigResponse) {
 	nginxConfigStatusMessage := configAppliedResponse
-	if response.configApply != nil {
-		if err := response.configApply.Complete(); err != nil {
-			nginxConfigStatusMessage = fmt.Sprintf("Config complete failed: %v", err)
-			log.Errorf(nginxConfigStatusMessage)
+	rollback := false
+
+	var rollbackError error
+	reloadErr := n.nginxBinary.Reload(response.nginxDetails.ProcessId, response.nginxDetails.ProcessPath)
+	if reloadErr != nil {
+		nginxConfigStatusMessage = fmt.Sprintf("Config apply failed (write): %v", reloadErr)
+		log.Errorf(nginxConfigStatusMessage)
+		rollbackError = reloadErr
+		rollback = true
+	} else {
+		rollbackError = n.monitor()
+		if rollbackError != nil {
+			nginxConfigStatusMessage = fmt.Sprintf("Config apply failed. The following errors were found in the NGINX error logs: %v", rollbackError)
```

Comment:
Need to update this message as it might not always be a NGINX error log error that is found. Could be a missing NGINX worker pid error.